### PR TITLE
[primtorch] `aten.normal` decomp has wrong return type due to `elementwise_type_promotion_wrapper`

### LINKED ISSUE #112449: [aot_autograd] torch.compile of torch.normal(.., dtype=torch.float64) ignores the dtype
### 🐛 Describe the bug

```python
import torch
for dtype in [torch.float64, torch.float16]:
    def fn():
        return torch.normal(2, 3, (100, 100), dtype=dtype, device="cpu")
    
    assert torch.compile(fn, backend="eager")().dtype ==  dtype # success
    assert torch.compile(fn, backend="aot_eager")().dtype == dtype  # success
    assert torch.compile(fn, backend="aot_eager_decomp_partition")().dtype == dtype  # fail
    assert torch.compile(fn, backend="inductor")().dtype == dtype  # fail
```

Also

### Versions

main

### Diagnosis

Seems like a prim decomp bug

cc: @bdhirsh @Chillee

### PR DESCRIPTION
**Pull request by jon-chuang**
Fixes https://github.com/pytorch/pytorch/issues/112449

elementwise_type_promotion_wrapper will promote `aten.normal` to the dtypes of `mean`, `std` args. 

But this is incorrect if we provide the dtype param. Hence, we allow overriding the result_dtype if a specified dtype arg is available.

This problem is unique to `aten.normal`, all other ops decorated do not have a dtype param.

cc @ezyang @mruberry @Lezcano @peterbell10 @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler

### CODE REVIEW

**Review by lezcano — COMMENTED** [review]
This can be done automatically. You can check if `"dtype"` is in the signature, and if so, that will take preference. No need to add a new arg

**Comment by jon-chuang** [comment]
> You can check if "dtype" is in the signature, and if so, that will take preference.

Yes, but I didn't want to override some other possible behaviour. Should it be opt in? `use_dtype_as_return_type=True`

**Comment by lezcano** [comment]
This is the desired behaviour whenever `dtype` is set, so no need for it to be opt in

**Comment by jon-chuang** [comment]
Ok, I made it use dtype arg by default, but possible to opt out.

File: torch/_prims_common/wrappers.py [hunk 1]
```diff
@@ -101,9 +104,11 @@ def __init__(
         *,
         type_promotion_kind: ELEMENTWISE_TYPE_PROMOTION_KIND,
         type_promoting_args: Optional[Sequence[str]] = None,
+        use_dtype_as_return_type: bool = True,
```
Comment (lezcano) [inline torch/_prims_common/wrappers.py:107]: The dtype behaviour is, or should be, consistent throughout PyToch.

Let's not add more arguments for the sake of adding them, things are already difficult enough as they are.

Comment (jon-chuang) [reply torch/_prims_common/wrappers.py:107]: removed

File: test/dynamo/test_functions.py [hunk 2]
```diff
@@ -2330,6 +2330,21 @@ def fn(x):
...
+                return torch.normal(2, 3, (10, 10), dtype=dtype, device="cpu")
+
+            self.assertEqual(
+                torch.compile(fn, backend="aot_eager_decomp_partition")().dtype,
+                dtype if dtype else torch.float32,
+            )
+            self.assertEqual(
+                torch.compile(fn, backend="inductor")().dtype,
+                dtype if dtype else torch.float32,
+            )
```
Comment (lezcano) [inline test/dynamo/test_functions.py:2346]: This is testing aot / inductor so it should go in the inductor folder.

**Review by lezcano — APPROVED** [review]
LGTM modulo a minor + CI passing.

File: test/dynamo/test_functions.py (hunk 2 above)
Comment (jon-chuang) [reply test/dynamo/test_functions.py:2346]: done

**Pull request closed without merging**